app/src/main/java/com/appdev/all_in_android/ui/navigation/NavigationStack.kt

```diff
@@ -0,0 +1,110 @@
+package com.appdev.all_in_android.ui.navigation
+
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.BottomNavigationItem
+import androidx.compose.material.Icon
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Home
+import androidx.compose.material3.NavigationBar
+import androidx.compose.material3.Scaffold
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.currentBackStackEntryAsState
+import androidx.navigation.compose.rememberNavController
+import com.appdev.all_in_android.ui.screens.HomeScreen
+import com.appdev.all_in_android.ui.screens.MarketplaceScreen
+import com.appdev.all_in_android.ui.screens.ProfileScreen
+
+@Composable
+fun NavigationSetup() {
+    val navController = rememberNavController()
+    val showBottomBar = rememberSaveable { mutableStateOf(false) }
+    val navBackStackEntry by navController.currentBackStackEntryAsState()
+
+    // For screens that do not display bottom navigation bar, change below
+    when (navBackStackEntry?.destination?.route) {
+        else -> {
+            showBottomBar.value = true
+        }
+    }
+
+    Scaffold(
+        bottomBar = {
+            if (showBottomBar.value) {
+                BottomNavigationBar(navController, NavigationItem.bottomNavTabList)
+            }
+        }
+    ) { innerPadding ->
+        SetupNavHost(
+            modifier = Modifier.padding(innerPadding),
+            navController = navController,
+            showBottomBar = showBottomBar
+        )
+    }
+}
+
+@Composable
+fun BottomNavigationBar(navController: NavHostController, tabItems: List<NavigationItem>) {
+    NavigationBar(
+        containerColor = Color.White,
+        contentColor = Color.Blue
```

Comment:
I think you can remove the contentColor = Color.Blue

Reply:
Other than that LGTM

Reply:
sounds good! lemme quickly fix then merge